flask_appbuilder/api/__init__.py

```diff
@@ -220,13 +220,37 @@ class ExampleApi(BaseApi):
                 base_permissions = ['can_get']
 
     """
+    class_permission_name = None
+    """
+        Override class permission name default fallback to self.__class__.__name__
+    """
+    previous_class_permission_name = None
```

Comment:
This could be a list but I don't think that is a necessity. If `class_permission_name` was to change twice over a few version, someone who have to make sure to bridge the versions.

Reply:
Yes, we need to be aware also that migrations can only go so far. If we converge two classes to the same name, it's not possible to automatically converge them back

Reply:
Actually I take that back, it's possible, this needs more work

Reply:
Done, we can now revert/expand permissions back after having compressing them, so forget my previous comment on not being possible to "automatically converge them back"